Protocol/contracts/BondFactory.sol

```diff
@@ -55,9 +56,9 @@ contract BondFactory {
             Subscriber memory subscriber = Subscriber(payable(msg.sender), subscriptionValue, 0);
             subscribers[_bondId][msg.sender] = subscriber;
         }
-        depositToFactory();
-        selectedBond.subscribeToBond(_subscriptionAmount);
-        emit SubscribedToBond(_bondId, msg.sender, subscriptionValue);
+        emitAddress(address(selectedBond));
+        selectedBond.subscribeToBond(_subscriptionAmount, msg.sender);
```

Comment:
Can the emit of the BondWallet bond address be followed straight away by the subscribeToBond call? Will the DAI approve function happen in time?